File: ui.frontend/src/components/Container/ReactMainContainer/ReactMainContainer.js
import React from 'react';
import StlyedContainer from './ReactMainContainer.styled';
import Title from '../../micro/Title/Title';
import Text from '../../micro/Text/Text';
import Button from '../../micro/Button/Button';

const ReactMainContainer = ({
    title,
    titleColor,
    text,
    textColor,
    button,
    buttonTextColor,
    buttonBkgdColor,
}) => {
    return (
        <StlyedContainer>
            <Title color={titleColor}>{title}</Title>
            <Text color={textColor}>{text}</Text>
            <Button color={buttonTextColor} background={buttonBkgdColor}>
                {button}
            </Button>
        </StlyedContainer>
    );
};

export default ReactMainContainer;

File: core/src/main/java/com/reactapp/core/models/ReactMainContainer.java
/*
 * ***********************************************************************
 * React App CONFIDENTIAL
 * ___________________
 *
 * Copyright 2022 React App.
 * All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains the property
 * of React App and its suppliers, if any. The intellectual and
 * technical concepts contained herein are proprietary to React App
 * and its suppliers and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from React App.
 * ***********************************************************************
 */

package com.reactapp.core.models;

import com.adobe.cq.export.json.ComponentExporter;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.osgi.annotation.versioning.ConsumerType;


/**
 * Defines the {@code ReactMainContainer} Sling Model used for the {@code reactapp/components/react-main-container} component.
 * 
 */
@ConsumerType
public interface ReactMainContainer
    extends ComponentExporter
{


    @JsonProperty("title")
    String getTitle();

    @JsonProperty("titleColor")
    String getTitleColor();

    @JsonProperty("text")
    String getText();

    @JsonProperty("textColor")
    String getTextColor();

    @JsonProperty("button")
    String getButton();

    @JsonProperty("buttonBkgdColor")
    String getButtonBkgdColor();

    @JsonProperty("buttonTextColor")
    String getButtonTextColor();

}
